# Midtest - Usecase Python (Titanic)

## By Firdaus Adi Nugroho

In [8]:
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings ('ignore')

# Dataset
Titanic dataset.
You can download it here.<br>
https://drive.google.com/open?id=15IyT1ODuDKgZb8WN6iG64hFJRZWJTSAz

# Backround
The dataset contains data for 887 of the real Titanic passengers. Each row represents
one person. The columns describe different attributes about the person including whether they
survived (S), their age (A), their passenger-class (C), their sex (G) and the fare they paid (X). The table
below shows the Data Dictionary.

# Objectives
Create a python script that contains the script to answer the questions below. Please put
comments on the script to show which line of codes answers which questions. Please upload the
script after you finish.

# Questions
1. What is the dimension (col, row) of the data frame?
2. How to know data type of each variable?
3. How many passengers survived (Survived=1) and not-survived (Survived=0)?
4. How to drop column ‘Name’ from the data frame?
5. Add one new column called ‘family’ to represent number of family-member aboard (hint: family = sibsp + parch)
6. As shown, columns ‘Age’ contains missing values. Please add new column named ‘Age_miss’ to indicate whether Age is missing or not (Age_miss = ‘YES’ for missing value and ‘NO’ for non-missing value).
7. Please fill Age missing value with means of existing Age values
8. What is the maximum passenger Age who survived from the tragedy?
9. How many passengers survived from each ‘PClass’?
10. How to randomly split the data frame into 2 parts (titanic1 and titanic2) with proportion of 0.7 for tttanic1 and 0.3 for titanic2 ?

In [3]:
#Meload dataset
df = pd.read_csv('titanic.csv')
df.head()

Survived  Pclass                                               Name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  
0    male  22.0                        1                        0   7.2500  
1  female  38.0                        1                        0  71.2833  
2  female  26.0                        0                        0   7.9250  
3  female  35.0                        1                        0  53.1000  
4    male  35.0                        0                        0   8.0500

### 1. What is the dimension (col, row) of the data frame?

In [4]:
#Melihat dimensi dari dataset

df.shape

(887, 8)

Dimensi dari dataset titanic ialah<br>
kolom: 8<br>
baris: 887

## 2. How to know data type of each variable?

In [5]:
#melihat informasi tipe data tiap variabel

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 887 entries, 0 to 886
Data columns (total 8 columns):
Survived                   887 non-null int64
Pclass                     887 non-null int64
Name                       887 non-null object
Sex                        887 non-null object
Age                        887 non-null float64
Siblings/Spouses Aboard    887 non-null int64
Parents/Children Aboard    887 non-null int64
Fare                       887 non-null float64
dtypes: float64(2), int64(4), object(2)
memory usage: 55.5+ KB


Setiap kolom memiliki tipe data, kita dapat mengetahui data info menggunakan syntax df.info() sehingga terdapat informasi data tipe setiap kolom

## 3. How many passengers survived (Survived=1) and not-survived (Survived=0)?

### Survived

In [6]:
survive = df[df['Survived']==1]
survive.head()

Survived  Pclass                                               Name  \
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
8         1       3   Mrs. Oscar W (Elisabeth Vilhelmina Berg) Johnson   
9         1       2                 Mrs. Nicholas (Adele Achem) Nasser   

      Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  
1  female  38.0                        1                        0  71.2833  
2  female  26.0                        0                        0   7.9250  
3  female  35.0                        1                        0  53.1000  
8  female  27.0                        0                        2  11.1333  
9  female  14.0                        1                        0  30.0708

Berikut adalah tampilan data penumpang yang selamat. Dilihat dari kolom 'Survived menunjukkan angka 1'

In [57]:
survive['Survived'].count()

342

Dari 887 penumpang titanic, terdapat 342 penumpang yang selamat. Berarti, tragedi titanic ini memakan korban lebih dari 50% meninggal dunia

### Unsurvived

In [58]:
unsurvive = df[df['Survived']==0]
unsurvive.head()

Survived  Pclass                           Name   Sex   Age  \
0         0       3         Mr. Owen Harris Braund  male  22.0   
4         0       3        Mr. William Henry Allen  male  35.0   
5         0       3                Mr. James Moran  male  27.0   
6         0       1         Mr. Timothy J McCarthy  male  54.0   
7         0       3  Master. Gosta Leonard Palsson  male   2.0   

   Siblings/Spouses Aboard  Parents/Children Aboard     Fare  
0                        1                        0   7.2500  
4                        0                        0   8.0500  
5                        0                        0   8.4583  
6                        0                        0  51.8625  
7                        3                        1  21.0750

In [59]:
unsurvive['Survived'].count()

545

Dari 887 penumpang titanic, terdapat 545 penumpang yang tidak terselamatkan, hal itu mengapa tragedi titanic ini sangat booming karena selain kapal terbesar pada zamannya, juga memakan korban yang sangat banyak

## 4. How to drop column ‘Name’ from the data frame?

In [60]:
df.drop(['Name'],axis=1, inplace=True)
df.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0         0       3    male  22.0                        1   
1         1       1  female  38.0                        1   
2         1       3  female  26.0                        0   
3         1       1  female  35.0                        1   
4         0       3    male  35.0                        0   

   Parents/Children Aboard     Fare  
0                        0   7.2500  
1                        0  71.2833  
2                        0   7.9250  
3                        0  53.1000  
4                        0   8.0500

Melakukan drop column "Name"

## 5. Add one new column called ‘family’ to represent number of family-member aboard (hint: family = sibsp + parch)

In [61]:
df['Family'] = df['Siblings/Spouses Aboard'] + df['Parents/Children Aboard']
df['Family'].head()

0    1
1    1
2    0
3    1
4    0
Name: Family, dtype: int64

## 6. As shown, columns ‘Age’ contains missing values. Please add new column named ‘Age_miss’ to indicate whether Age is missing or not (Age_miss = ‘YES’ for missing value and ‘NO’ for non-missing value).

In [63]:
df['Ages_miss'] = 'null'
df['Ages_miss'][df['Age'] == None] = 'Yes'
df['Ages_miss'][df['Age'] != None] = 'No'
df.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0         0       3    male  22.0                        1   
1         1       1  female  38.0                        1   
2         1       3  female  26.0                        0   
3         1       1  female  35.0                        1   
4         0       3    male  35.0                        0   

   Parents/Children Aboard     Fare  Family Ages_miss  
0                        0   7.2500       1        No  
1                        0  71.2833       1        No  
2                        0   7.9250       0        No  
3                        0  53.1000       1        No  
4                        0   8.0500       0        No

## 7. Please fill Age missing value with means of existing Age values

In [64]:
df.isnull().sum()

Survived                   0
Pclass                     0
Sex                        0
Age                        0
Siblings/Spouses Aboard    0
Parents/Children Aboard    0
Fare                       0
Family                     0
Ages_miss                  0
dtype: int64

Mengecek apakah ada missing value pada setiap kolom

In [67]:
df['Age'] = df['Age'].fillna(df['Age'].mean())

Mengisi missing value kolom Age menggunakan data rata-rata 'Age'. Sehingga data yang kosong akan terisi rata-rata umur pada dataset

## 8. What is the maximum passenger Age who survived from the tragedy?

In [69]:
survive = df[df['Survived']==1]
agemax = survive['Age'].max()
print(agemax)

80.0


## 9. How many passengers survived from each ‘PClass’?

In [72]:
survive = df[df['Survived']==1]
survive_P = survive.groupby('Pclass')
survive_Pclass = survive_P['Pclass'].count()
print(survive_Pclass)

Pclass
1    136
2     87
3    119
Name: Pclass, dtype: int64


Kita bisa lihat dari data diatas, peredaran penumpang yang menempati kamar-kamar berdasarkan kelasnya. Kamar kelas 1 merupakan kelas penumpang yang paling banyak selamat, lalu kelas kamar 2 merupakan kelas yang paling sedikit selamat

## 10. How to randomly split the data frame into 2 parts (titanic1 and titanic2) with proportion of 0.7 for tttanic1 and 0.3 for titanic2 ?

In [73]:
from sklearn.model_selection import train_test_split
x_train ,x_test = train_test_split(df,test_size=0.3)

Memisahkan dataframe menjadi dataframe x_train(70% dari dataset, dan x_test(30& dari dataset). hal itu ditunjukkan dari berapa test_size yang ingin kita tentukan sebelumnya

In [74]:
x_train.shape

(620, 9)

x_train memiliki jumlah sebagai berikut:<br>
Kolom: 9<br>
Baris: 620

In [75]:
x_test.shape

(267, 9)

x_train memiliki jumlah sebagai berikut:<br>
Kolom: 9<br>
Baris: 267

# Kesimpulan
Kesimpulan dari data ini ialah bagaimana kita mengetahui informasi-informasi secara real dari tragedi tenggelamnya titanic dari dataset titanic ini. Terdapat 50% lebih penumpan titanic meninggal dunia, dan korban yang paling banyak korban ialah berasal dari kamar golongan 2